packages/flutter_tools/lib/src/compile.dart

```diff
@@ -0,0 +1,48 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:convert' show JSON;
+import 'dart:io' as io show File;
+
+import 'package:flutter_tools/src/artifacts.dart';
+import 'package:flutter_tools/src/base/common.dart';
+import 'package:flutter_tools/src/globals.dart';
+import 'package:front_end/compiler_options.dart';
+import 'package:front_end/kernel_generator.dart';
+import 'package:kernel/ast.dart';
+import 'package:kernel/kernel.dart';
+import 'package:kernel/target/flutter_fasta.dart';
+import 'package:kernel/target/targets.dart';
+
+Map<String, Uri> loadDartLibraries() {
+  final Uri libraries = new Uri.file(artifacts.getArtifactPath(Artifact.platformLibrariesJson));
+  final dynamic map = JSON.decode(new io.File.fromUri(libraries).readAsStringSync())['libraries'];
+  final Map<String, Uri> dartLibraries = <String, Uri>{};
+  map.forEach((String k, String v) => dartLibraries[k] = libraries.resolve(v));
+  return dartLibraries;
+}
+
+Future<String> compile({String packagesPath, String mainPath}) async {
+  final String platformKernelDill = artifacts.getArtifactPath(
+      Artifact.platformKernelDill);
+  // TODO(aam): Move FlutterFastaTarget to flutter tools.
+  final CompilerOptions options = new CompilerOptions()
+    ..dartLibraries = loadDartLibraries()
+    ..packagesFileUri = Uri.parse(packagesPath)
+    ..sdkSummary = Uri.parse(platformKernelDill)
+    ..linkedDependencies = <Uri>[Uri.parse(platformKernelDill)]
```

Comment:
Ah, I see.

We need to iterate on this part of the API. Maybe we should inspect linked dependencies before we try to infer a summary. I'll file a bug on our end to look further

Meanwhile, consider to replace:
```dart
..sdkSummary = ...
```
with
```
..compileSdk = true
```

When `compileSdk` is true, we operate in a lazy-fashion: only when we see an import to a `dart:*` library, we will try to find out where the code is coming from. If it is already in the linkedDependencies, then we don't do anything else. The 'uimplemented' error you saw would only come up if you happen to import a dart-library not included in the linked dependencies (which is not possible in your case).